modules/hivemq-edge-module-http/src/main/java/com/hivemq/edge/adapters/http/HttpAdapterConfig.java

```diff
@@ -120,59 +87,93 @@ public enum HttpContentType {
-    private boolean httpPublishSuccessStatusCodeOnly = true;
+    private final boolean httpPublishSuccessStatusCodeOnly;
 
     @JsonProperty("allowUntrustedCertificates")
     @ModuleConfigField(title = "Allow the adapter to read from untrusted SSL sources (for example expired certificates).",
                        defaultValue = "false",
                        format = ModuleConfigField.FieldType.BOOLEAN)
-    private boolean allowUntrustedCertificates = false;
+    private final boolean allowUntrustedCertificates;
 
     @JsonProperty("assertResponseIsJson")
     @ModuleConfigField(title = "Assert JSON Response?",
                        description = "Always attempt to parse the body of the response as JSON data, regardless of the Content-Type on the response.",
                        defaultValue = "false",
                        format = ModuleConfigField.FieldType.BOOLEAN)
-    private boolean assertResponseIsJson = false;
-
-    public HttpAdapterConfig() {
-    }
-
-    public HttpAdapterConfig(final @NotNull String adapterId) {
-        this.id = adapterId;
+    private final boolean assertResponseIsJson;
+
+    @JsonCreator
+    public HttpAdapterConfig(
+            @JsonProperty(value = "id", required = true) final @NotNull String id,
+            @JsonProperty("pollingIntervalMillis") @JsonAlias("publishingInterval") final @Nullable Integer pollingIntervalMillis,
+            @JsonProperty("maxPollingErrorsBeforeRemoval") final @Nullable Integer maxPollingErrorsBeforeRemoval,
+            @JsonProperty(value = "url", required = true) final @NotNull String url,
+            @JsonProperty(value = "destination", required = true) final @NotNull String destination,
+            @JsonProperty("qos") final @Nullable Integer qos,
+            @JsonProperty("httpRequestMethod") final @Nullable HttpMethod httpRequestMethod,
+            @JsonProperty("httpRequestBodyContentType") final @Nullable HttpContentType httpRequestBodyContentType,
+            @JsonProperty("httpRequestBody") final @Nullable String httpRequestBody,
+            @JsonProperty("httpConnectTimeout") final @Nullable Integer httpConnectTimeoutSeconds,
+            @JsonProperty("httpHeaders") final @Nullable List<HttpHeader> httpHeaders,
+            @JsonProperty("httpPublishSuccessStatusCodeOnly") final @Nullable Boolean httpPublishSuccessStatusCodeOnly,
+            @JsonProperty("allowUntrustedCertificates") final @Nullable Boolean allowUntrustedCertificates,
+            @JsonProperty("assertResponseIsJson") final @Nullable Boolean assertResponseIsJson) {
+        this.id = id;
+        if (pollingIntervalMillis != null) {
+            this.pollingIntervalMillis = pollingIntervalMillis;
+        } else {
+            this.pollingIntervalMillis = 1000;
+        }
+        this.maxPollingErrorsBeforeRemoval = Objects.requireNonNullElse(maxPollingErrorsBeforeRemoval, 10);
+        this.url = url;
```

Comment:
while the url is marked with required, I'm not sure of this is 100% not null. I wonder if we want to check that it is not null. Same for the other parameters.